### DISCUSSION TITLE
스프링 테스트에서 `@Autowired` 어노테이션을 달아줘야만 객체 주입이 되는 이유를 설명해주세요.

### DISCUSSION THREAD

**Discussion by gyungchan-jo** [2024-woowacourse-study/level-interview#225] [Spring Test]
.

**Comment by alstn113** [comment]
SpringApplication의 경우 Bean을 주입해주는 역할을 스프링이 담당한다. 하지만 테스트의 경우 JUnit에 의해 객체를 주입하게 된다. 이 때 JUnit은 ParameterResolver를 사용해서 주입을 한다.

Parameter Resolver는 테스트에 동적으로 parameter를 주입할 수 있도록 JUnit에서 제공하는 인터페이스이다. 사용하려면 `@ExtendWith(xxxExtension.class)`를 테스트 위에 붙여야 한다.

`@SpringBootTest`, `@JdbcTest` 내부에는 `@ExtendWith(SpringExtension.class)`이 등록되어 있으며, SpringExtension은 ParameterResolver를 구현한 구현체다. SpringExtension 클래스의 supportParamter 메서드에서는 스프링 컨테이너에서 가져올 수 있는 빈 타입 여부를 `@AutoWired` 어노테이션이 있는지를 가지고 판단한다. 

Junit에서 `@Autowired`를 사용하지 않으면 SpringExtension은 해당 타입의 객체를 제공할 수 없다고 응답하게 되고, Junit은 해당 타입을 처리할 ParameterResolver가 없다고 에러를 발생시키게 된다.

참고: https://forkyy.tistory.com/62

**Comment by 3Juhwan** [comment]
와 진짜 참고자료가 윗 코멘트와 동일해서 더 달 말이 없습니다.
